### ISSUE TITLE
add extra env variables to container spec that allow `valueFrom`

### ISSUE THREAD

**Issue by Ashtonian** [BeeInventor/charts#15]
I'm [using](https://github.com/Ashtonian/charts/commit/44b289a25e469a74f9cc0890391216c908c671dc) host_ip injected as a env variable to then be referenced in nsqd.args as a statsd endpoint. However I realize not everyone needs/wants this injected. Is there a way to make this easily configurable in the yaml? 

Goal syntax would be something like: 

container spec:
```yaml
containers:
    env:
   {{- if .Values.env }}
   {{ toYaml .Values.env }}
   {{- end }}
``` 

values.yaml: 
```yaml
env:
- name: SOME_ENV
  value: someval 
- name: SOME_FIELD_ENV
  valueFrom:
      fieldRef:
          fieldPath: status.podIP 
```

When I try to use the above I get formatting errors and `Invalid YAML: mapping values are not allowed in this context:`. My yaml foo is lacking, do you have any suggestions for how to do this? I can submit a pr.

**Comment by benjamin658** [comment]
Hi @Ashtonian,  please try 

```yaml
{{- if Values.env }}
{{- toYaml .Values.env | nindent 2 }}
{{- end -}}
```

**Comment by benjamin658** [comment]
The extraEnv is included in the chart version v1.2.1.
